middleware/http/oauth2clientcredentials/oauth2clientcredentials_middleware.go

```diff
@@ -98,27 +102,38 @@ func (m *Middleware) GetHandler(_ context.Context, metadata middleware.Metadata)
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			var headerValue string
 
-			// Check if valid token is in the cache
-			cachedToken, found := m.tokenCache.Get(cacheKey)
-			if !found {
-				m.log.Debugf("Cached token not found, try get one")
-
-				token, err := m.tokenProvider.GetToken(r.Context(), conf)
-				if err != nil {
-					m.log.Errorf("Error acquiring token: %s", err)
+			if meta.pathFilterRegex != nil {
+				matched := meta.pathFilterRegex.MatchString(r.URL.Path)
+				if !matched {
+					m.log.Debugf("PathFilter %s didn't match %s! Skipping!", meta.PathFilter, r.URL.Path)
+					next.ServeHTTP(w, r)
 					return
 				}
+			}
 
-				tokenExpirationDuration := time.Until(token.Expiry)
-				m.log.Debugf("Token expires at %s (%s from now)", token.Expiry, tokenExpirationDuration)
-
-				headerValue = token.Type() + " " + token.AccessToken
-				m.tokenCache.Set(cacheKey, headerValue, tokenExpirationDuration)
-			} else {
-				m.log.Debugf("Cached token found for key %s", cacheKey)
+			// Check if valid token is in the cache
+			cachedToken, found := m.tokenCache.Get(cacheKey)
+			if found {
+				m.log.Infof("Cached token found for key %s", cacheKey)
 				headerValue = cachedToken.(string)
+				r.Header.Add(meta.HeaderName, headerValue)
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			m.log.Infof("Cached token not found, try get one")
```

Comment:
```suggestion
			m.log.Debugf("Cached token not found, try get one")
```
also, can you make the ending clearer?

Reply:
done!